envoy/stream_info/address_set_accessor.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/network/address.h"
+#include "envoy/stream_info/filter_state.h"
+
+namespace Envoy {
+namespace StreamInfo {
+
+/**
+ * A FilterState object that holds a set of network addresses.
+ */
+class AddressSetAccessor : public FilterState::Object {
+public:
+  virtual void add(Network::Address::InstanceConstSharedPtr address) PURE;
+
+  virtual void
+  iterate(const std::function<bool(const Network::Address::InstanceConstSharedPtr& address)>&)
```

Comment:
javadoc comments for envoy/ functions if you leave them here

Alternately looking at another example filterSate object we just use the impl everywhere and don't have an abstract base class.  I think that'd be fine here too, especially if we add a key() function so all the files using the class have to include the Impl anyway